packages/mongodb-constants/src/filter.ts

```diff
@@ -0,0 +1,201 @@
+   */
+  meta?: (Meta | 'field:*' | 'accumulator:*' | 'expr:*')[];
+  /**
+   * Stage-only filters
+   */
+  stage?: StageFilterOptions;
+};
+
+function matchesMeta(filter: string[], meta: string) {
+  const metaParts = meta.split(':');
+  return filter.some((metaFilter) => {
+    const filterParts = metaFilter.split(':');
+    return (
+      filterParts.length === metaParts.length &&
+      filterParts.every((part, index) => {
+        return part === '*' || part === metaParts[index];
+      })
+    );
+  });
+}
+
+function isIn<T extends string | number>(
+  val: T | T[] | undefined,
+  set: T[] | undefined
+): boolean {
+  // Do not filter when either value or match set is not provided
+  if (typeof val === 'undefined' || typeof set === 'undefined') {
+    return true;
+  }
+  // Otherwise check that value intersects with the match set
+  val = Array.isArray(val) ? val : [val];
+  return val.some((v) => set.includes(v));
+}
+
+export function createConstantFilter({
+  meta: filterMeta,
+  serverVersion = '999.999.999',
+  stage: filterStage = {},
+}: Pick<FilterOptions, 'meta' | 'serverVersion' | 'stage'> = {}): (
+  completion: Completion
+) => boolean {
+  const currentServerVersion =
+    /^(?<version>\d+?\.\d+?\.\d+?)/.exec(serverVersion)?.groups?.version ??
+    serverVersion;
+  return ({ version: minServerVersion, meta, env, namespace, apiVersions }) => {
+    return (
+      gte(currentServerVersion, minServerVersion) &&
+      isIn(filterStage.env, env) &&
+      isIn(filterStage.namespace, namespace) &&
+      isIn(filterStage.apiVersion, apiVersions) &&
+      (!filterMeta || matchesMeta(filterMeta, meta))
+    );
+  };
+}
+
+function isValidIdentifier(identifier: string) {
+  // Quick check for common case first
+  if (/[.\s"'()[\];={}:]/.test(identifier)) {
+    return false;
+  }
+  try {
+    // Everything else we check using eval as regex methods of checking are quite
+    // hard to do (see https://mathiasbynens.be/notes/javascript-identifiers-es6)
+    // eslint-disable-next-line @typescript-eslint/no-implied-eval
+    new Function(`"use strict";let _ = { ${identifier}: 0 };`);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Helper method to conditionally wrap value if it's not a valid identifier
+ */
+export function wrapField(field: string, force = false): string {
+  return force || !isValidIdentifier(field)
+    ? `"${field.replace(/["\\]/g, '\\$&')}"`
+    : field;
+}
+
+function normalizeField(
+  field: string | { name: string; description?: string }
+) {
+  return typeof field === 'string'
+    ? { value: field }
+    : {
+        value: field.name,
+        description: field.description,
+      };
+}
+
+/**
+ * Convenience method to filter list of mongodb constants based on constants
+ * values
+ *
+ * @param options filter options
+ * @param constants list of constants to filter, for testing purposes only
+ * @returns filtered constants
+ */
+export function filter(
```

Comment:
I actually like the previous `completer` name more, because it not only filters methods, it returns the completion values. Maybe it can be `getFilteredComplitions` or something to point that it has a filter under the hood but also stores values?

Reply:
Sure, renamed. I think having a prefix match was making it more of a `completer`, now that this is removed, it's not really like that, so I went with `getFilteredCompletions` as it's somewhere in the middle